akka-sample-cqrs-scala/src/main/scala/sample/cqrs/EventProcessor.scala

```diff
@@ -35,41 +33,30 @@ import org.slf4j.LoggerFactory
  */
 object EventProcessor {
 
-  case object Ping extends CborSerializable
-
-  def entityKey(eventProcessorId: String): EntityTypeKey[Ping.type] = EntityTypeKey[Ping.type](eventProcessorId)
-
   def init[Event](
       system: ActorSystem[_],
       settings: EventProcessorSettings,
       eventProcessorStream: String => EventProcessorStream[Event]): Unit = {
-    val eventProcessorEntityKey = entityKey(settings.id)
-
-    ClusterSharding(system).init(Entity(eventProcessorEntityKey)(entityContext =>
-      EventProcessor(eventProcessorStream(entityContext.entityId))).withRole("read-model"))
-
-    KeepAlive.init(system, eventProcessorEntityKey)
+      val shardedDaemonSettings = ShardedDaemonProcessSettings(system)
+          .withKeepAliveInterval(settings.keepAliveInterval)
+          .withShardingSettings(ClusterShardingSettings(system).withRole("read-model"))
+      ShardedDaemonProcess(system)
+        .init[Nothing](s"event-processors-${settings.id}", settings.parallelism, i => EventProcessor(eventProcessorStream(s"${settings.tagPrefix}-$i")), shardedDaemonSettings, None)
   }
 
-  def apply(eventProcessorStream: EventProcessorStream[_]): Behavior[Ping.type] = {
-
-    Behaviors.setup { context =>
+  def apply(eventProcessorStream: EventProcessorStream[_]): Behavior[Nothing] = {
+    Behaviors.setup[Nothing] { ctx  =>
+      ctx.log.info("Event processor running {}", eventProcessorStream)
```

Comment:
toString of the `eventProcessorStream` is probably not that pretty?

Reply:
👍